test/connectors/tinyman/tinyman.test.ts

```diff
@@ -0,0 +1,258 @@
+        clawback_address: '',
+        liquidity_in_usd: '744662.849801994861',
+        last_day_volume_in_usd: '58.407348762305',
+        last_week_volume_in_usd: '2261.387003578427',
+        last_day_price_change: '-0.156310',
+        is_stable: false,
+        is_wrapped: false,
+      },
+      {
+        id: USDC_TOKEN_ID.toString(),
+        is_liquidity_token: false,
+        name: 'USDC',
+        unit_name: USDC_TOKEN,
+        decimals: USDC_DECIMALS,
+        total_amount: '18446744073709551615',
+        url: 'https://centre.io',
+        is_verified: true,
+        clawback_address:
+          'XM2W7VZODABS6RAL3FENBRKCOF6XLOQZZWIVVZTBYCVH2ADRYKN53CQLXM',
+        liquidity_in_usd: '210464.272543000000',
+        last_day_volume_in_usd: '8.000000000000',
+        last_week_volume_in_usd: '6198.073873000000',
+        last_day_price_change: '0.000000',
+        is_stable: false,
+        is_wrapped: false,
+      },
+    ];
+  });
+};
+
+beforeAll(async () => {
+  algorand = Algorand.getInstance(NETWORK);
+  await algorand.init();
+  patchCurrentBlockNumber();
+  patchGetAssetData();
+
+  tm = Tinyman.getInstance(NETWORK);
+  await tm.init();
+});
+
+afterEach(() => {
+  unpatch();
+});
+
+afterAll(async () => {
+  await algorand.close();
+});
+
+const patchSendHelper = () => {
+  patch(Swap.v2, 'execute', () => {
+    return { txID: 1234, round: 1 };
+  });
+};
+
+const patchFetcher = () => {
+  patch(poolUtils.v2, 'getPoolInfo', () => {
+    return {
+      account: {
+        lsig: {
+          tag: new Uint8Array([80, 114, 111, 103, 114, 97, 109]),
+          logic: new Uint8Array([
+            6, 128, 24, 0, 0, 0, 0, 59, 193, 147, 29, 0, 0, 0, 0, 1, 225, 171,
+            112, 0, 0, 0, 0, 0, 0, 0, 0, 129, 0, 91, 53, 0, 52, 0, 49, 24, 18,
+            68, 49, 25, 129, 1, 18, 68, 129, 1, 67,
+          ]),
+          args: undefined,
+          sig: undefined,
+          msig: undefined,
+        },
+        sigkey: undefined,
+      },
+      validatorAppID: 1002541853,
+      asset1ID: 31566704,
+      asset2ID: 0,
+      status: 'ready',
+      contractVersion: 'v2',
+      asset1ProtocolFees: BigInt('10961420982'),
+      asset2ProtocolFees: BigInt('48196942141'),
+      asset1Reserves: BigInt('1200097723737'),
+      asset2Reserves: BigInt('6633822022882'),
+      issuedPoolTokens: BigInt('2690871970933'),
+      cumulativePriceUpdateTimeStamp: 1682547803,
+      protocolFeeRatio: 6,
+      totalFeeShare: BigInt('30'),
+      poolTokenID: 1002590888,
+    };
+  });
+};
+
+describe('verify Tinyman estimate Sell Trade', () => {
+  it('Should return an ExpectedTrade when available', async () => {
+    patchFetcher();
+
+    const expectedTrade = await tm.estimateTrade({
+      chain: CHAIN_NAME,
+      network: NETWORK,
+      base: 'ALGO',
+      quote: 'USDC',
+      amount: '1',
+      side: 'SELL',
+    });
+    expect(expectedTrade).toHaveProperty('trade');
+    expect(expectedTrade).toHaveProperty('expectedAmount');
```

Comment:
We should check the values of at least `expectedAmount` and `expectedPrice` here

Reply:
done